presto-hive/src/main/java/io/prestosql/plugin/hive/DeleteDeltaLocations.java

```diff
@@ -0,0 +1,197 @@
+    }
+
+    public static class DeleteDeltaInfo
+    {
+        private final long minWriteId;
+        private final long maxWriteId;
+        private final int statementId;
+
+        @JsonCreator
+        public DeleteDeltaInfo(
+                @JsonProperty("minWriteId") long minWriteId,
+                @JsonProperty("maxWriteId") long maxWriteId,
+                @JsonProperty("statementId") int statementId)
+        {
+            this.minWriteId = minWriteId;
+            this.maxWriteId = maxWriteId;
+            this.statementId = statementId;
+        }
+
+        @JsonProperty
+        public long getMinWriteId()
+        {
+            return minWriteId;
+        }
+
+        @JsonProperty
+        public long getMaxWriteId()
+        {
+            return maxWriteId;
+        }
+
+        @JsonProperty
+        public int getStatementId()
+        {
+            return statementId;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            DeleteDeltaInfo that = (DeleteDeltaInfo) o;
+            return minWriteId == that.minWriteId &&
+                    maxWriteId == that.maxWriteId &&
+                    statementId == that.statementId;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(minWriteId, maxWriteId, statementId);
+        }
+
+        @Override
+        public String toString()
+        {
+            return toStringHelper(this)
+                    .add("minWriteId", minWriteId)
+                    .add("maxWriteId", maxWriteId)
+                    .add("statementId", statementId)
+                    .toString();
+        }
+    }
+
+    public static Builder builder(Path partitionPath)
+    {
+        return new Builder(partitionPath);
+    }
+
+    public static class Builder
+    {
+        private final String partitionLocation;
+        private final ImmutableList.Builder<DeleteDeltaInfo> deleteDeltaInfoBuilder;
+
+        private Builder(Path partitionPath)
+        {
+            requireNonNull(partitionPath, "partitionPath is null");
+            partitionLocation = partitionPath.toString();
+            deleteDeltaInfoBuilder = ImmutableList.builder();
+        }
+
+        public void addDeleteDelta(Path deleteDeltaPath, long minWriteId, long maxWriteId, int statementId)
+        {
+            requireNonNull(deleteDeltaPath, "deleteDeltaPath is null");
+            String partitionPathFromDeleteDelta = deleteDeltaPath.getParent().toString();
+            checkArgument(
+                    partitionLocation.equals(partitionPathFromDeleteDelta),
+                    format("Partition location in DeleteDelta, %s, does not match stored location %s", deleteDeltaPath.getParent().toString(), partitionLocation));
```

Comment:
* `%s` -> `'%s'`
* don't use `format` inside `checkArgument`; `checkArgument` will do this for you